File: src/main/java/nuclearcoder/discordbot/database/SqlSingletons.java
package nuclearcoder.discordbot.database;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class SqlSingletons {

    public static final void ensureExists() throws SQLException
    {
        PreparedStatement statement = Database.conn.prepareStatement("SELECT `_` FROM `singleton`");

        ResultSet rs = statement.executeQuery();
        boolean hasFirst = rs.first();
        rs.close();
        if (!hasFirst)
        {
            statement.executeUpdate(
                    "INSERT INTO `singleton` (`_`, `cahchannel`, `cahdecks`, `cahconfig`) VALUES (0, '', '', '')");
        }

        statement.close();
    }

}

File: src/main/java/nukebot/database/SqlSingletons.kt
package nukebot.database

import java.sql.SQLException

object SqlSingletons {

    @Throws(SQLException::class)
    fun ensureExists() {
        val statement = Database.conn?.prepareStatement("SELECT `_` FROM `singleton`") ?: return

        val rs = statement.executeQuery()
        val hasFirst = rs.first()
        rs.close()
        if (!hasFirst)
            statement.executeUpdate(
                    "INSERT INTO `singleton` (`_`, `cahchannel`, `cahdecks`, `cahconfig`) VALUES (0, '', '', '')")

        statement.close()
    }

}
